shm/Map.cpp

```diff
@@ -23,7 +23,6 @@ void Map::generateIslands(int numOfIslandsToGenerate) {
                 return island.getPosition() == Coordinates(positionX, positionY); 
             })) { 
             addIsland({positionX, positionY});
-            --numOfIslandsToGenerate;
         } 
     } while (--numOfIslandsToGenerate); 
 }
```

Comment:
What would you say about this? It might be easier to understand what's going on there. 
```cpp
void Map::generateIslands(size_t numOfIslandsToGenerate) {
    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_int_distribution<> distrib(minPositionXY, maxPositionXY);
    
    size_t numOfGeneratedIslands = 0;
    size_t positionX {}; 
    size_t positionY {};
    do { 
        positionX = distrib(gen); 
        positionY = distrib(gen); 
        if (std::none_of(islandsLocations_.begin(), islandsLocations_.end(), 
                [positionX, positionY](const auto& island) { 
                return island.getPosition() == Coordinates(positionX, positionY); 
            })) { 
            addIsland({positionX, positionY});
            ++numOfGeneratedIslands;
        } 
    } while (numOfGeneratedIslands != numOfIslandsToGenerate); 
}
```

OR


```cpp
void Map::generateIslands(size_t numOfIslandsToGenerate) {
    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_int_distribution<> distrib(minPositionXY, maxPositionXY);
    
    size_t positionX {}; 
    size_t positionY {};
    do { 
        positionX = distrib(gen); 
        positionY = distrib(gen); 
        if (std::none_of(islandsLocations_.begin(), islandsLocations_.end(), 
                [positionX, positionY](const auto& island) { 
                return island.getPosition() == Coordinates(positionX, positionY); 
            })) { 
            addIsland({positionX, positionY});
            --numOfIslandsToGenerate;
        } 
    } while (numOfIslandsToGenerate != 0); 
}
```

Reply:
Ok, I'm fixing this now. 
I can do this later (rectangle-shaped map), after PR, no problem :P